### DISCUSSION TITLE
Error: convert-im6.q16: improper image header

### DISCUSSION THREAD

**Discussion by lucamancusodev** [ImageMagick/ImageMagick#4656] [Help] [answered]
Hi,
I'm trying to convert a multi channel psd file, however every time ImageMagick returns: "_convert-im6.q16: improper image header"_

Even using: "identify -verbose" I got the same error.

I can open the file with photshop without any issue.
[test.zip](https://github.com/ImageMagick/ImageMagick/files/7802771/test.zip)
The psd file is attached.

**Comment by snibgo** [comment]
What version of IM, on what platform?

With IM v7.1.0-4 and v7.1.0-20 on Windows 8.1, your file can be read without complaint. It takes some time, probably because of the large quantity of metadata.

**Comment by lucamancusodev** [comment]
IM: 6.9.10-23 Q16 x86_64 20190101
OS: Debian 9 - Linux 4.19.208-1

**Comment by dlemstra** [comment, marked as answer]
This is happening because we have a different check inside IM6 than we have in IM7. I just pushed a patch to make sure this is fixed in the next release of ImageMagick 6.

**Reply by lucamancusodev** [reply]
Hi, do you knwo when I can update so I get the fix?

**Reply by dlemstra** [reply]
We will probably publish a new release some time this week.

**Reply by lucamancusodev** [reply]
Hi, from debian buster packages I see the 6.9.10.23 version which is the same as mine.
When you think it will be released?

I use IM for my customers but I can't convert some images.

**Reply by dlemstra** [reply]
We don't know. We don't maintain the buster packages.

**Reply by lucamancusodev** [reply]
What version do you maintain? Or where I can get the updated IM?

**Reply by dlemstra** [reply]
You will need to either build ImageMagick from source or wait for the maintainers of the buster packages to update their packages.

**Reply by lucamancusodev** [reply]
Hi, I have successful compiled and installed the newer version (7.1.0-21) from source, but now when I try to convert the image it returns: `"convert: delegate library support not built-in 'test.psd' (XML) @ warning/profile.c/ValidateXMPProfile/1945."`

probably because I don't installed the xml delegate support but I can't figure out how to install it, tried to compile with `--with-xml=yes` but is the same....

Thank you!

**Reply by dlemstra** [reply]
You will also need to install the dependencies. When you do a `configure` at the end it will print what is being used. You will probably need to install `libxml2`.

**Reply by lucamancusodev** [reply]
I have yet installed `libxml2`, this is what I have when `configure `ends.

![immagine](https://user-images.githubusercontent.com/41821623/151119105-1141987b-e675-4444-9b8f-08395200a3fc.png)

**Comment by lucamancusodev** [comment]
Thank you, I will update when it comes out.
